packages/terra-application-utility/src/header/_HeaderUtilityMenu.jsx

```diff
@@ -0,0 +1,256 @@
+      return (
+        <ul className={cx('body')}>
+          {currentItem.childKeys.map((key) => {
+            let item = null;
+            if (this.getItem(key).contentLocation === Utils.LOCATIONS.BODY) {
+              item = this.buildItem(key);
+            }
+            return item;
+          })}
+        </ul>
+      );
+    }
+    return null;
+  }
+
+  buildFooterContent(currentItem) {
+    if (currentItem && currentItem.childKeys && currentItem.childKeys.length) {
+      return currentItem.childKeys.map((key) => {
+        let item = null;
+        if (this.getItem(key).contentLocation === Utils.LOCATIONS.FOOTER) {
+          item = this.buildItem(key);
+        }
+        return item;
+      });
+    }
+    return null;
+  }
+
+  /**
+   * Function to trigger when an item is selected.
+   * 1. Has children: navigate to the next page
+   * 2. Toggles: trigger onChange without closing the menu.
+   * 3. Endpoint: close menu and trigger onChange.
+   * @param {*} event
+   * @param {*} key
+   */
+  handleOnChange(event, key) {
+    const childKeys = this.getItem(key).childKeys;
+    const item = this.getItem(key);
+    if (typeof childKeys !== 'undefined' && childKeys.length > 0) {
+      this.toggleIsSelected(key);
+      this.setState({
+        previousKey: this.push(this.state.currentKey),
+        currentKey: key,
+      });
+    } else {
+      this.toggleIsSelected(key);
+      if (item.isSelectable !== true) {
+        this.props.onRequestClose();
+      }
+      this.props.onChange(event, key);
+    }
+  }
+
+  handleRequestBack() {
+    this.setState({ currentKey: this.pop() });
+  }
+
+  pop() {
+    const newStack = this.state.previousKeyStack.slice();
+    const lastKey = newStack.pop();
+    this.setState({ previousKeyStack: newStack });
+    return lastKey;
+  }
+
+  push(key) {
+    const newStack = this.state.previousKeyStack.slice();
+    newStack.push(key);
+    this.setState({ previousKeyStack: newStack });
+  }
+
+  toggleIsSelected(key) {
+    const item = this.getItem(key);
+    item.isSelected = !item.isSelected;
+    this.state.map.set(key, item);
+  }
+
+  render() {
+    const {
+      menuItems,
+      initialSelectedKey,
+      isHeightBounded,
+      onChange,
+      onRequestClose,
+      ...customProps
+    } = this.props;
+
+    const MenuClassNames = cx(['utility-menu', customProps.classNames]);
+    const headerClassNames = cx('header');
+    const leftContainerClassNames = cx('left-container');
+    const backButtonClassNames = cx('back-button');
+    const iconLeftClassNames = cx('icon-left');
+    const headerTextClassNames = cx('header-text');
+    const footerClassNames = cx('footer');
+    const footerDividerClassNames = cx('footer-divider');
+
+    const currentKey = this.state.currentKey;
+    const currentItem = this.getItem(currentKey);
+    const backButton = (
+      <button className={backButtonClassNames} onClick={this.handleRequestBack} aria-label={'Back button'}>
```

Comment:
Aria label needs to be i18n'd, look at action header for the translations for 'back'.